src/DotVVM.Framework/Compilation/ControlTree/ControlTreeResolverBase.cs

```diff
@@ -137,6 +137,12 @@ protected virtual void ResolveRootContent(DothtmlRootNode root, IAbstractContent
                 return null;
             }
             var viewmodelDirective = (IAbstractViewModelDirective)directives[ParserConstants.ViewModelDirectiveName].First();
+            if (viewmodelDirective.ResolvedType.IsAssignableTo(new ResolvedTypeDescriptor(typeof(IDotvvmControl))))
```

Comment:
I'd check for `DotvvmBindableObject`. However, we should mainly check that in the serializer - it does seem quite likely that someone uses a control inside the viewmodel.